PWGJE/Tasks/crCharmHadronization.cxx

```diff
@@ -0,0 +1,159 @@
+#include "Framework/runDataProcessing.h"
+
+struct crCharmHadronizationTask {
+  // producing new table
+  Produces<aod::JetDistanceTable> distJetTable;
+
+  // Histogram registry: an object to hold your histograms
+  HistogramRegistry registry{"histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  // histogram configurables
+  Configurable<int> distBins{"distBins", 1000, "number of bins in distance histogram"};
+
+  void init(InitContext const&)
+  {
+
+    // create histograms
+    // D0 candidate histograms
+    registry.add("h_jet_counter", ";# jets;", {HistType::kTH1F, {{2, 0., 1.}}});
+    registry.add("h_d0_jet_projection", ";z^{D^{0},jet}_{||};dN/dz^{D^{0},jet}_{||}", {HistType::kTH1F, {{distBins, 0., 10.}}});
+    registry.add("h_d0_jet_distance_vs_projection", ";#DeltaR_{D^{0},jet};z^{D^{0},jet}_{||}", {HistType::kTH2F, {{distBins, 0., 10.}, {distBins, 0., 10.}}});
+    registry.add("h_d0_jet_distance", ";#DeltaR_{D^{0},jet};dN/d(#DeltaR)", {HistType::kTH1F, {{distBins, 0., 10.}}});
+    registry.add("h_d0_jet_pt", ";p_{T,D^{0} jet};dN/dp_{T,D^{0} jet}", {HistType::kTH1F, {{200, 0., 10.}}});
+    registry.add("h_d0_jet_eta", ";#eta_{T,D^{0} jet};dN/d#eta_{D^{0} jet}", {HistType::kTH1F, {{250, -5., 5.}}});
+    registry.add("h_d0_jet_phi", ";#phi_{T,D^{0} jet};dN/d#phi_{D^{0} jet}", {HistType::kTH1F, {{250, -10., 10.}}});
+    registry.add("h_d0_mass", ";m_{D^{0}} (GeV/c^{2});dN/dm_{D^{0}}", {HistType::kTH1F, {{1000, 0., 10.}}});
+    registry.add("h_d0_eta", ";#eta_{D^{0}} (GeV/c^{2});dN/d#eta_{D^{0}}", {HistType::kTH1F, {{250, -5., 5.}}});
+    registry.add("h_d0_phi", ";#phi_{D^{0}} (GeV/c^{2});dN/d#phi_{D^{0}}", {HistType::kTH1F, {{250, -10., 10.}}});
+  }
+
+  void processDummy(aod::TracksIU const& tracks) {}
+  PROCESS_SWITCH(crCharmHadronizationTask, processDummy, "Dummy process function turned on by default", false);
+
+  void processDataChargedSubstructure(JetCollision const& collision,
+                                      soa::Join<aod::D0ChargedJets, aod::D0ChargedJetConstituents> const& jets,
+                                      JetTracks const& tracks,
+                                      CandidatesD0Data const& d0cands)
+  {
+    // CandidatesLcData const& lccands) {
+
+    double axisDistance = 0;
+
+    for (auto& jet : jets) {
+      // fill jet counter histogram
+      registry.fill(HIST("h_jet_counter"), 0.5);
+      // obtaining jet 3-vector
+      TVector3 jetVector(jet.px(), jet.py(), jet.pz());
+
+      for (auto& d0Candidate : jet.hfcandidates_as<CandidatesD0Data>()) { // for jet constituents use -> auto& jetConstituent : jet.tracks_as<JetTracks>()
+
+        // obtaining jet 3-vector
+        TVector3 d0Vector(d0Candidate.px(), d0Candidate.py(), d0Candidate.pz());
+        // calculating fraction of the jet momentum carried by the D0 along the direction of the jet axis
+        double z_parallel = (jetVector * d0Vector) / (jetVector * jetVector);
```

Comment:
is this operand the dot product or the cross product?

Reply:
It's the scalar dot product